src/main/java/at/hannibal2/skyhanni/utils/renderables/Renderable.kt

```diff
@@ -244,21 +279,86 @@ interface Renderable {
             return Collections.singletonList(string(string))
         }
 
-        fun string(string: String) = object : Renderable {
+        fun string(
+            text: String,
+            horizontalAlign: HorizontalAlignment = HorizontalAlignment.Left,
+            verticalAlign: VerticalAlignment = VerticalAlignment.Top,
+        ) = object : Renderable {
             override val width: Int
-                get() = Minecraft.getMinecraft().fontRendererObj.getStringWidth(string)
+                get() = Minecraft.getMinecraft().fontRendererObj.getStringWidth(text)
             override val height = 10
+            override val horizontalAlign = horizontalAlign
+            override val verticalAlign = verticalAlign
 
             override fun render(posX: Int, posY: Int) {
-                Minecraft.getMinecraft().fontRendererObj.drawStringWithShadow("§f$string", 1f, 1f, 0)
+                Minecraft.getMinecraft().fontRendererObj.drawStringWithShadow("§f$text", 1f, 1f, 0)
             }
         }
 
-        fun placeholder(width: Int) = object : Renderable {
-            override val width: Int = width
-            override val height = 10
+        fun placeholder(width: Int, height: Int = 10) = object : Renderable {
+            override val width = width
+            override val height = height
+            override val horizontalAlign = HorizontalAlignment.Left
+            override val verticalAlign = VerticalAlignment.Top
+
+            override fun render(posX: Int, posY: Int) {
+            }
+        }
+
+        fun texturedProgressBar(
+            percent: Float,
+            color: Color = Color(0, 255, 0),
+            useChroma: Boolean = false,
+            width: Int = 30,
+            height: Int = 4,
+            horizontalAlign: HorizontalAlignment = HorizontalAlignment.Left,
+            verticalAlign: VerticalAlignment = VerticalAlignment.Top,
```

Comment:
change after the merge with #887 
```suggestion
            horizontalAlign: HorizontalAlignment = HorizontalAlignment.LEFT,
            verticalAlign: VerticalAlignment = VerticalAlignment.TOP,
```